File: 2025/01/python/sol-p1.py
import os
import sys

def debug(*kargs):
    if os.getenv("DEBUG") == "1":
        print(*kargs)


def main():
    assert len(sys.argv) == 2, "Usage: python3 sol-p1.py filename"
    
    cur_position = 50
    count_zero = 0

    debug("Input file:", sys.argv[1])

    with open(sys.argv[1], "r") as input_file:
        for instruction in input_file:
            instruction = instruction.strip()
            debug(instruction)

            direction = instruction[0]
            distance = int(instruction[1:])
            debug(direction, distance)

            distance = distance % 100
            if direction == 'L':
                distance = -distance
            debug(cur_position, distance)
            
            new_position = (cur_position + distance + 100) % 100

            if new_position == 0:
                count_zero += 1

            cur_position = new_position

    print("Password: ", count_zero)


if __name__ == "__main__":
    main()

File: 2025/01/go/sol-p1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

var isDebug = os.Getenv("DEBUG") == "1"

func main() {
	if len(os.Args) != 2 {
		log.Fatal("Error: Usage: go run sol-p1.go file")
	}

	fileName := os.Args[1]
	if isDebug { fmt.Println("Input file:", fileName) }

	file, err := os.Open(fileName)
	if err != nil {
		log.Fatal("Failed to read file:", err)
	}

	curPosition := 50
	countZero := 0

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		instruction := scanner.Text()
		if isDebug { fmt.Println(instruction) }
		
		direction := instruction[:1]
		distance, err := strconv.Atoi(instruction[1:]) 
		if err != nil {
			log.Fatal("Failed to parse distance:", instruction[1:])
		}
		if isDebug { fmt.Println(direction, distance) }

		distance %= 100
		if direction == "L" {
			distance = -distance
		}
		
		newPosition := (curPosition + distance + 100) % 100
		if isDebug { fmt.Println(curPosition, distance, newPosition) }

		if newPosition == 0 {
			countZero++
		}

		curPosition = newPosition
	}

	file.Close()

	fmt.Println("Password:", countZero)
}
